platform-dependent-components/problem-solver/cxx/verification-module/agent/not-enough-specified-objects-search-agent.cpp

```diff
@@ -0,0 +1,211 @@
+/*
+ * This source file is part of an OSTIS project. For the latest info, see http://ostis.net
+ * Distributed under the MIT License
+ * (See accompanying file COPYING.MIT or copy at http://opensource.org/licenses/MIT)
+ */
+
+#include <fstream>
+#include <filesystem>
+
+#include <sc-memory/sc_keynodes.hpp>
+#include "sc-agents-common/utils/IteratorUtils.hpp"
+
+#include "config/config.hpp"
+
+#include "constants/verification_constants.hpp"
+
+#include "keynodes/verification_keynodes.hpp"
+
+#include "not-enough-specified-objects-search-agent.hpp"
+
+using namespace utils;
+
+namespace verificationModule
+{
+
+std::filesystem::path SearchNotEnoughSpecifiedObjectsAgent::filePath;
+
+ScResult SearchNotEnoughSpecifiedObjectsAgent::DoProgram(ScAction & action)
+{
+  // list of  <templAddr, elements>
+  std::vector<std::pair<ScAddr, ScAddrVector>> templAddrAndElementsList = GetSpecificationTemplateAndElements();
+
+  std::stringstream infoAboutBasicSpec;
+  std::string nodeName;
+  std::string subjectDomainName;
+
+  filePath = Config::getInstance()->getValue(FileConfigs::VERIFICATION_ENDPOINT, FileConfigs::FILE_PATH);
+
+  std::for_each(
+      std::begin(templAddrAndElementsList),
+      std::end(templAddrAndElementsList),
+      [&](std::pair<ScAddr, ScAddrVector> & templAddrAndElements)
+      {
+        ScAddr const & templAddr = templAddrAndElements.first;
+        ScAddrVector const & elementsAddrs = templAddrAndElements.second;
+        std::for_each(
+            std::begin(elementsAddrs),
+            std::end(elementsAddrs),
+            [&](ScAddr const & elementAddr)
+            {
+              try
+              {
+                SC_LOG_INFO(m_context.GetElementSystemIdentifier(elementAddr));
+                CheckTemplate(m_context, elementAddr, templAddr);
+              }
+              catch (utils::ScException const & exception)  // No template or no specification
```

Comment:
I think that cases with no template and unsuccessful search by template are different
and there is a question about whether to print first case in a file, that contains not enough specified objects